contracts/PriceCalculator.sol

```diff
@@ -0,0 +1,160 @@
+     */
+    function _getBuyPrice(
+        uint256 _multipleFactor,
+        uint256 _growthFactor,
+        uint256 _constantFactor,
+        uint256 _numOfOutstandingTokens
+    ) internal pure returns (uint256) {
+        uint256 currTokenIdx = _numOfOutstandingTokens.add(1);
+        uint256 subFormulaCalc = _subFormulaPower(_growthFactor, currTokenIdx);
+        uint256 price = _constantFactor.add(_multipleFactor.mul(subFormulaCalc));
+        return price;
+    }
+
+    /**
+     * @dev Similar to _getPrice(), but calculates the price of a purchase quantity more than 1.
+     * @param _multipleFactor           The multiplicative factor that adjusts the ‘slope’ of the curve (aka 'm').
+     * @param _growthFactor             The % growth in price per factor-increase (aka 'c', between 1 and 100).
+     * @param _numOfOutstandingTokens   The number of follower tokens outstanding (aka 'x').
+     * @param _constantFactor           The constant that adjusts the functions in the spirit of y = mx + b (aka 'b').
+     * @param _amountToBuy              The amount of tokens to calculate the price for according to the bonding curve.
+     * @return price The current price of the creator's token.
+     */
+    function _getBatchBuyPrice(
+        uint256 _multipleFactor,
+        uint256 _growthFactor,
+        uint256 _constantFactor,
+        uint256 _numOfOutstandingTokens,
+        uint256 _amountToBuy
+    ) internal pure returns (uint256) {
+        uint256 totalConstant = _amountToBuy * _constantFactor;
+        uint256 totalNonConstant;
+
+        // Save recurring integer overflow checking on `i++` in for loop (due to Solidity 0.8)
+        unchecked {
+            for (uint256 i = 1; i <= _amountToBuy; i++) {
+                uint256 currTokenIdx = _numOfOutstandingTokens.add(i);
+                uint256 subFormulaCalc = _subFormulaPower(_growthFactor, currTokenIdx);
+                totalNonConstant = totalNonConstant.add(subFormulaCalc);
+            }
+        }
+
+        uint256 price = totalConstant.add(_multipleFactor.mul(totalNonConstant));
+        return price;
+    }
+
+    function _getSellPrice(
+        uint256 _multipleFactor,
+        uint256 _growthFactor,
+        uint256 _constantFactor,
+        uint256 _numOfOutstandingTokens
+    ) internal pure returns (uint256) {
+        uint256 subFormulaCalc = _subFormulaPower(_growthFactor, _numOfOutstandingTokens);
+        uint256 price = _constantFactor.add(_multipleFactor.mul(subFormulaCalc));
+        return price;
+    }
+
+    function _getBatchSellPrice(
+        uint256 _multipleFactor,
+        uint256 _growthFactor,
+        uint256 _constantFactor,
+        uint256 _numOfOutstandingTokens,
+        uint256 _amountToSell
+    ) internal pure returns (uint256) {
+        uint256 totalConstant = _amountToSell * _constantFactor;
+        uint256 totalNonConstant;
+
+        unchecked {
+            for (uint256 i = 0; i < _amountToSell; i++) {
```

Comment:
Is it possible to get rid of this loop[call: `_subFormulaPower` only once]. If it is.
Then we can reduce gas consumption.